nautobot/docs/plugins/development.md

```diff
@@ -574,6 +575,134 @@ A `PluginMenuButton` has the following attributes:
 !!! note
     Any buttons associated within a menu item will be shown only if the user has permission to view the link, regardless of what permissions are set on the buttons.
 
+## Navigation Menu Tabs
+
+Plugins can modify the existing navigation bar layout by using `nav_tabs` inside of `navigation.py`. Using the key and weight system, a developer can modify existing tabs, groups, items and buttons.
+
+### Adding a new tab
+
+The code below shows how to add a new tab to the navbar. A tab is defined by a `NavMenuTab` object. Similarly a group is defined using `NavMenuGroup`. Both of these objects are used as containers for actual items.
+
+The position in the navigation menu is defined by the weight. The lower the weight the closer to the same the object will be. All core objects have weights in multiples of 100., meaning there is plenty of space around the objects for plugins to customize.
+
+Below you can see `Dummy Tab` has a weight value of `150`. This means the tab will appear between `Organization` and `Devices`.
```

Comment:
Do we have any straightforward way to introspect the current menu tab/group/item/button weights so as to figure out what value a plugin *should* use, short of digging through the code and reading each individual `navigation.py` file?

As an idea, maybe extend the nav-menu rendering template to add the weight of each tab/group/item/button as an HTML element [data attribute](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) that could be inspected via the browser's dev-tools?

Reply:
Nice idea

Reply:
Thanks for adding that. Mind also adding some notes here about how this information is available and can be used to figure out the desired weight?

Reply:
Ah, I see you added that below already  -thanks! Might be better to move it up to here though?

Reply:
will move now

Reply:
actually does it need to be moved if we are now storing navbar documentation in `navigation-menu.md`?

Reply:
See my new comments. :-)